prov/rxd/src/rxd_cq.c

```diff
@@ -137,8 +133,17 @@ static void rxd_complete_rx(struct rxd_ep *ep, struct rxd_x_entry *rx_entry)
 			FI_WARN(&rxd_prov, FI_LOG_EP_CTRL, "could not write error entry\n");
 			return;
 		}
+		goto out;
 	}
 
+	if (rx_entry->cq_entry.flags & FI_REMOTE_CQ_DATA ||
+	    (!(rx_entry->flags & RXD_NO_RX_COMP) &&
+	      rx_entry->cq_entry.flags & FI_RECV))
+		rx_cq->write_fn(rx_cq, &rx_entry->cq_entry);
+
+	assert(rx_entry->op < ofi_op_last);
+	ofi_ep_rx_cntr_inc_funcs[rx_entry->op](&ep->util_ep);
```

Comment:
Hmm... moving this assertion into a common location would be best.  We could define:
```
static inline ofi_ep_rx_cntr_inc(struct util_ep *ep, u8 opcode)
{
    assert(opcode < ofi_op_last)
    ofi_ep_rx_cntr_inc_funcs[opcode](ep);
}
```
add only access the function array through that.

Reply:
btw - this could be added as a separate change